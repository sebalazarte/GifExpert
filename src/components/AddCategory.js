import React, { useState, Fragment } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({ setCategorias, setLimite }) => { //se desestructuro props

    const [inputValue, setInputValue] = useState('');
    const [numberValue, setNumberValue] = useState(10);

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleNumberChange = (e) => {
        setNumberValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategorias(cats => [inputValue, ...cats,]);
            setInputValue('');
        }

        if (numberValue.length > 0) {
            setLimite(numberValue);
        }
    }

    const handleClean = (e) => {
        e.preventDefault();
        setCategorias([]);
    }

    return (
        <Fragment>
            <form onSubmit={handleSubmit}>
                <div className="row g-3">
                    <div className="col-md-8">
                        <div className="input-group mb-3">
                            <label className="input-group-text">Buscar</label>
                            <input
                                id="txtBuscar"
                                type="text"
                                value={inputValue}
                                onChange={handleInputChange}
                                className="form-control"
                                placeholder="ingrese para buscar..."
                                required />
                            <label className="input-group-text">Limite</label>
                            <input
                                id="txtCantidad"
                                type="number"
                                value={numberValue}
                                onChange={handleNumberChange}
                                className="form-control cantidad"
                            />
                            <button type="submit" className="btn btn-primary">
                                Buscar
                            </button>
                        </div>
                    </div>
                    <div className="col-md-4">
                        <button onClick={handleClean} className="btn btn-danger">
                            Limpiar
                        </button>   
                    </div>
                </div>
            </form>

        </Fragment>
    )
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired
}