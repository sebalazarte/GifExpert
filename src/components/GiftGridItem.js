import React from 'react'

export const GiftGridItem = ({id, title, url}) => {
    // console.log(props);
    return (
        <div className="card animate__animated animate__bounceIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
