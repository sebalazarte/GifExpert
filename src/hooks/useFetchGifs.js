import { useState, useEffect } from 'react';
import { getGifts } from '../helpers/getGifs';

export const useFetchGifs = (category, limit) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifts(category, limit)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                });
            });
    }, [category, limit]);

    return state;
}