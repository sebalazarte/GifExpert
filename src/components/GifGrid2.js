import React, { useState, useEffect, Fragment } from 'react'
import { getGifts } from '../helpers/getGifs';
import { GiftGridItem } from './GiftGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect(()=>{
        getGifts(category)
            .then( setImages );
    }, [category]);

    return (
        <Fragment>
            <h3>{category}</h3>
            <div className="card-grid">

                {
                    images.map(img =>
                        <GiftGridItem
                            key={img.id}
                            {...img} />)
                }
            </div>
        </Fragment>
    )
}
