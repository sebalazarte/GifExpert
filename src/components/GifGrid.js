import React, {Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category, limit }) => {

    const {data: images, loading} = useFetchGifs(category, limit);

    return (
        <Fragment>
            <h3 className="animate__animated animate__faden">{category}</h3>
            
            {loading && <p className="animate__animated animate__flash">Cargando...</p>}

            <div className="card-grid">

                {
                    images.map(img =>
                        <GifGridItem
                            key={img.id}
                            {...img} />)
                }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limit: PropTypes.number.isRequired
}