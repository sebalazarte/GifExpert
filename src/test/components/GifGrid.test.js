import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
import '@testing-library/jest-dom';

describe('Pruebas en el componente <GifGrid/>', () => {

    const category = 'goku';
    const limit = "10";

    test('Debe mostrar el control correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} limit={limit} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imagenes en useFetchGifs', () => {


        const gifs = [
            {
                id: '123456',
                title: 'Dragon Ball GIF by TOEI Animation UK',
                url: 'https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=70a38d38casbobveplnboyjb4ahtzdhdxe3dfv9dztcr6q6d&rid=giphy.gif&ct=g'
            },
            {
                id: '123456',
                title: 'Dragon Ball GIF by TOEI Animation UK',
                url: 'https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=70a38d38casbobveplnboyjb4ahtzdhdxe3dfv9dztcr6q6d&rid=giphy.gif&ct=g'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} limit={limit} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

});