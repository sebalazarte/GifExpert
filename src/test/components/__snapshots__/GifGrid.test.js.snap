// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid/> Debe mostrar el control correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__faden"
  >
    goku
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid/> Debe mostrar items cuando se cargan imagenes en useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__faden"
  >
    goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123456"
      key="123456"
      title="Dragon Ball GIF by TOEI Animation UK"
      url="https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=70a38d38casbobveplnboyjb4ahtzdhdxe3dfv9dztcr6q6d&rid=giphy.gif&ct=g"
    />
    <GifGridItem
      id="123456"
      key="123456"
      title="Dragon Ball GIF by TOEI Animation UK"
      url="https://media1.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=70a38d38casbobveplnboyjb4ahtzdhdxe3dfv9dztcr6q6d&rid=giphy.gif&ct=g"
    />
  </div>
</Fragment>
`;
